### PR TITLE
[DCOS-51019] Support seccomp in SDK

### PR DESCRIPTION
**Pull request by kvish**

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawPod.java [hunk 1]
```diff
@@ -79,6 +85,16 @@ private RawPod(
     this.sharePidNamespace = sharePidNamespace != null && sharePidNamespace;
     this.allowDecommission = allowDecommission != null && allowDecommission;
     this.hostVolumes = hostVolumes == null ? new WriteOnceLinkedHashMap<>() : hostVolumes;
+    this.seccompUnconfined = seccompUnconfined;
+    this.seccompProfileName = seccompProfileName;
+    validateSeccomp();
+  }
+
+  private void validateSeccomp() throws Exception{
+    if (seccompProfileName != null && seccompUnconfined != null) {
```
Comment (kensipe) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/RawPod.java]: this validation is incomplete...    It is necessary to validate not only that `seccompUnconfined` != null but also that if `seccompUnconfined` ==  true that `seccompProfileName` is null 

`if (seccompProfileName != null && (seccompUnconfined != null && seccompUnconfined)) {`

and we should have a test for this condition.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java [hunk 2]
```diff
@@ -573,10 +573,27 @@ private static String getConfigTemplateDownloadPath(ConfigFileSpec config) {
...
       containerInfo.addVolumes(Protos.Volume.newBuilder()
           .setContainerPath("/tmp")
           .setHostPath("tmp")
           .setMode(Protos.Volume.Mode.RW));
+
+      LOGGER.info("Setting seccomp info unconfined: {} profile: {}",
+              podSpec.getSeccompUnconfined(),
+              podSpec.getSeccompProfileName());
+
+      containerInfo.getLinuxInfoBuilder().setSeccomp(Protos.SeccompInfo.newBuilder()
```
Comment (kaiwalyajoshi) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java]: I'd prefer to explicitly encode the XOR behaviour as mentioned under the second bullet under 'Notes' [here](https://jira.mesosphere.com/browse/DCOS-49133?focusedCommentId=254417&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-254417).

For the case where `podSpec.getSeccompProfileName().isPresent()` we can explicitly set 
`Protos.SeccompInfo.newBuilder().setUnconfined(false).build()` to be unambiguous an in full compliance with [this](https://github.com/apache/mesos/blob/c6f3334c43e95f6486942256f77ed2aa926c4d1b/include/mesos/mesos.proto#L3222).

Comment (kvish) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java]: this should work as is already. if seccompProfileName is defined, seccompUnconfined is defaulted to false so if seccompProfileName is passed we will always be setting unconfined to false, which is why Unconfined is not an optional and has no if condition above it in PodInfoBuilder

File: frameworks/helloworld/tests/test_seccomp.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+
+@pytest.fixture(scope="module", autouse=True)
+def configure_package(configure_security):
+    try:
+        # setup custom seccomp profile
+        all_agent_ips = set([agent["hostname"] for agent in sdk_agents.get_agents()])
+        for ip in all_agent_ips:
+            print("Copying custom seccomp profile to agent %s" % ip)
```
[the remark this answers is not in the document]
Comment (kvish) [reply frameworks/helloworld/tests/test_seccomp.py]: ya nice for logging to debug the test

File: frameworks/helloworld/tests/test_seccomp.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+        sdk_install.install(config.PACKAGE_NAME, config.SERVICE_NAME, NUM_HELLO, additional_options=options)
+
+        yield  # let the test session execute
+    finally:
+        sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+
+
+@pytest.mark.dcos_min_version("1.13")
+@pytest.mark.sanity
```
[the remark this answers is not in the document]
Comment (kvish) [reply frameworks/helloworld/tests/test_seccomp.py]: its intended that dcos-enterprise CI will have seccomp enabled

File: frameworks/helloworld/tests/test_seccomp.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    marathon_config["env"]["HELLO_SECCOMP_PROFILE_NAME"] = "test_profile.json"
+    sdk_marathon.wait_for_deployment(config.SERVICE_NAME, 60, None)
+
+    @retrying.retry(
+        wait_fixed=1000,
+        stop_max_delay=120 * 1000,
+        retry_on_exception=lambda e: isinstance(e, Exception),
+    )
+    def check_tasks_fail():
+        assert sdk_tasks.get_failed_task_count(config.SERVICE_NAME) > 0
```
[the remark this answers is not in the document]
Comment (kvish) [reply frameworks/helloworld/tests/test_seccomp.py]: added call, accidentally removed it, also verified scp is idempotent I ran test 3 times and it consistently worked.

**Pull request merged**